### PR TITLE
HomeApp: rebuild webClips UI without clear and build

### PR DESCRIPTION
**Pull request by TomaszLukawskiSam**
[Issue] https://github.com/Samsung/TAU/issues/1454
[Problem] method `updateWebClipsUI` cause blinking
[Solution]
     - method has been rebuild to update items
    
Signed-off-by: Tomasz Lukawski <t.lukawski@samsung.com>

### CODE REVIEW

File: examples/mobile/HomeApp/js/app.js [hunk 1]
```diff
@@ -392,9 +430,7 @@ import Storage from "./clipping-storage.js";
 		tau.engine.createWidgets(webclipsContainer);
 	}
 
-	//TODO: provide mechanism for getting web clip name from webClip meta data
-	//		and separate from getting ID
-	function getWebClipName(webClip) {
+	function getWebClipNameFromUrl(webClip) {
```
Comment (lslachciak) [inline examples/mobile/HomeApp/js/app.js:436]: I think we should leave comment about getting webclipName from metadata and spearate it from getting webclip ID
we can have ID duplication problems in future

File: examples/mobile/HomeApp/js/app.js [hunk 2]
```diff
@@ -349,17 +349,38 @@ import Storage from "./clipping-storage.js";
 	}
 
 	function updateWebClipsUI() {
-		var current = document.querySelectorAll(".ui-card"),
-			webclipsContainer = document.getElementById("web-clips");
+		var webclipsContainer = document.getElementById("web-clips"),
+			currentWebClipsCards = [].slice.call(webclipsContainer.querySelectorAll(".ui-card[data-url],.ui-card[data-src]")),
```
Comment (lslachciak) [inline examples/mobile/HomeApp/js/app.js:354]: [].slice.call is not obvious,
please add some comment about this code here.

**Review by lslachciak — COMMENTED** [review]
I tested PR with HomeAppServer .
It works good. no blinking when active app is changed.
Some minor comments in code

File: examples/mobile/HomeApp/js/app.js (hunk 1 above)
Comment (TomaszLukawskiSam) [reply examples/mobile/HomeApp/js/app.js:436]: ok

File: examples/mobile/HomeApp/js/app.js (hunk 2 above)
Comment (TomaszLukawskiSam) [reply examples/mobile/HomeApp/js/app.js:354]: Done

**Review by lslachciak — APPROVED** [review]

File: examples/mobile/HomeApp/js/app.js [hunk 3]
```diff
@@ -369,7 +391,24 @@ import Storage from "./clipping-storage.js";
 			});
 		});
 
-		// activate / deactivate mini control cards
+		// set proper order of cards
+		// @todo change inline styles to css class after merge HomeApp branches
+		webclipsContainer.style.display = "flex";
+		webclipsContainer.style.flexDirection = "column";
+		currentWebClipsCards = [].slice.call(webclipsContainer.querySelectorAll(".ui-card[data-url],.ui-card[data-src]"));
```
Comment (singa2000) [inline examples/mobile/HomeApp/js/app.js:398]: Just I have a question.
I know about the "data-src" of ui-card. It's for card resource path like "webclip/apps-on-tv".
What's for use "data-url"?

Comment (TomaszLukawskiSam) [reply examples/mobile/HomeApp/js/app.js:398]: Hi,
Cards may be in two states one before load of card content (`data-src`) and after load of content (`data-url`). We need check both attributes.

Comment (singa2000) [reply examples/mobile/HomeApp/js/app.js:398]: ok. I see. Thank you for your kind explanation.

**Review by singa2000 — APPROVED** [review]

**Pull request merged**